update go tools

devshun opened the pull request:
close #553 

updated the Go tools used in `features/go`

## After

```
GO_TOOLS="\
    golang.org/x/tools/gopls@latest \
    honnef.co/go/tools/cmd/staticcheck@latest \
    golang.org/x/lint/golint@latest \
    github.com/mgechev/revive@latest \
    github.com/go-delve/delve/cmd/dlv@latest \
    github.com/fatih/gomodifytags@latest \
    github.com/haya14busa/goplay/cmd/goplay@latest \
    github.com/cweill/gotests/... \ 
    github.com/josharian/impl@latest"
```

## Description

### Additions

- **gomodifytags**
- **goplay**
- **gotests**
- **impl**

added the following tools that were newly added to [go tools list](https://github.com/golang/vscode-go/blob/v0.38.0/src/goToolsInformation.ts)

### Removed

- **gopkgs**
removed it because removed from the list in [go tools list](https://github.com/golang/vscode-go/blob/v0.38.0/src/goToolsInformation.ts)

- **go-outlines**
removed it because it had `replacedByGopls` set to true.

### Other

In addition, regarding linting tools, the following ones are not used by default, but they are set as optional attributes by the [official documentation](https://github.com/golang/vscode-go/blob/master/docs/tools.md#staticcheck), allowing users to choose them. Therefore, in this pull request, we have kept them without removing:

- golint
- revive
- go-clilint

samruddhikhandale commented on src/go/install.sh:
```diff
@@ -191,15 +191,18 @@ else
-    github.com/uudashr/gopkgs/v2/cmd/gopkgs@latest \
-    github.com/ramya-rao-a/go-outline@latest \
-    github.com/go-delve/delve/cmd/dlv@latest"
+    github.com/go-delve/delve/cmd/dlv@latest \
```
Can we add unit tests to validate these tools are installed fine? Thanks! 

Ref: https://github.com/devcontainers/features/blob/main/test/go/test.sh
https://github.com/devcontainers/cli/blob/main/docs/features/test.md

devshun replied:
@samruddhikhandale 

I've added tests for the tools we included this time. It appears that these tools don't have` ~~ --version` implemented for referencing the version, so I've only written tests for installation paths.

```
// test/go/test.sh
...

# gomodifytags
check "gomodifytags is installed at correct path" bash -c "which gomodifytags | grep /go/bin/gomodifytags"

# goplay
check "goplay is installed at correct path" bash -c "which goplay | grep /go/bin/goplay"

# gotests
check "gotests is installed at correct path" bash -c "which gotests | grep /go/bin/gotests"

# impl
check "impl is installed at correct path" bash -c "which impl | grep /go/bin/impl"

...
```

samruddhikhandale replied:
Thank you!

samruddhikhandale commented on src/go/install.sh:
```diff
@@ -191,15 +191,18 @@ else
 
 # Install Go tools that are isImportant && !replacedByGopls based on
-# https://github.com/golang/vscode-go/blob/v0.31.1/src/goToolsInformation.ts
+# https://github.com/golang/vscode-go/blob/v0.38.0/src/goToolsInformation.ts
```
@bamurtaugh/ @Chuxel any concerns with syncing the Go tools with latest https://github.com/golang/vscode-go/blob/v0.38.0/src/goToolsInformation.ts ?

Chuxel replied:
No concerns from my end - that's really the source of truth.

samruddhikhandale commented on src/go/install.sh:
```diff
@@ -191,15 +191,18 @@ else
 
 # Install Go tools that are isImportant && !replacedByGopls based on
-# https://github.com/golang/vscode-go/blob/v0.31.1/src/goToolsInformation.ts
+# https://github.com/golang/vscode-go/blob/v0.38.0/src/goToolsInformation.ts
```
Can we bump the minor version? 

https://github.com/devcontainers/features/blob/5a3c0b3f2f91997a2ffb05527d671a594a41d96d/src/go/devcontainer-feature.json#L3

devshun replied:
I have updated the minor version as follows:

```
// src/go/devcontainer-feature.json
...
 "version": "1.2.0",
...
```

samruddhikhandale reviewed (commented):
Thank you, left some comments!

devshun commented on src/go/install.sh:
```diff
@@ -191,15 +191,18 @@ else
+    github.com/go-delve/delve/cmd/dlv@latest \
+    github.com/fatih/gomodifytags@latest \
+    github.com/haya14busa/goplay/cmd/goplay@latest \
+    github.com/cweill/gotests/... \ 
```
@samruddhikhandale 

Thank you for your advice! My apologies, I made a mistake, github.com/cweill/gotests/... was the installation path for go get. As you've pointed out, I've corrected it to use the @latest flag. After making the changes, I was also able to confirm that the tests are successful!

```
 github.com/cweill/gotests/gotests@latest
```

devshun commented:
@samruddhikhandale 

Thank you for your review! I have addressed your comments. When you have a moment, could you please check again?

- [x] https://github.com/devcontainers/features/pull/559#discussion_r1204527924
- [x] https://github.com/devcontainers/features/pull/559#discussion_r1204529194
- [x] https://github.com/devcontainers/features/pull/559#discussion_r1204531155

devshun commented:
@microsoft-github-policy-service agree

samruddhikhandale reviewed (approved):
Looks good, thanks!
